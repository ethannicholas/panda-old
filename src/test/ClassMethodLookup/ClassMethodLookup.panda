class ClassMethodLookup {
    method test() {
        process1("Hello")
        process2("Goodbye")
    }

    method process1(o:Object) {
        Console.writeLine("process1 instance method")
    }

    @class
    method process1(s:String) {
        Console.writeLine("process1 class method")
    }

    method process2(s:String) {
        Console.writeLine("process2 instance method")
    }

    @class
    method process2(o:Object) {
        Console.writeLine("process2 class method")
    }

    @class
    method main() {
        ClassMethodLookup().test()
    }
}