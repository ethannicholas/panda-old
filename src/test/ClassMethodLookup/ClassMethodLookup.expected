process1 class method
process2 instance method
